litmus/litmus.c

```diff
@@ -314,6 +316,50 @@ asmlinkage long sys_null_call(cycles_t __user *ts)
 	return ret;
 }
 
+/*	This is a helper syscall to query if the budget has been exhausted.
+ *
+ *	returns 0 on successful query
+ *	returns EPERM if task is not a real-time task.
+ *      returns EFAULT if &job is not a valid pointer.
+ */
+asmlinkage long sys_query_budget_exhausted(int __user *flag)
+{
+	int retval = -EPERM;
+	if (is_realtime(current))
+		retval = put_user(budget_exhausted(current), flag);
+
+	return retval;
+}
+
+/*	This is a helper syscall to query the slack before
+ *  the deadline.
+ *
+ *	returns 0 on successful query
+ *	returns EPERM if task is not a real-time task.
+ *      returns EFAULT if &job is not a valid pointer.
+ */
+asmlinkage long sys_query_slack_time(lt_t __user *exec_time)
+{
+	int retval = -EPERM;
+	lt_t currTime = litmus_clock();
+	lt_t deadline = current->rt_param.job_params.deadline;
+
+	if (is_realtime(current))
+	{
+		// retval = put_user(budget_remaining(current), exec_time);
+		if(currTime < deadline)
+		{
+			retval = put_user(deadline - currTime, exec_time);
```

Comment:
Also, if you need precise control over deadlines, you may want to use `clock_nanosleep()` to trigger job wake-ups.